lib/deserialize.js

```diff
@@ -49,6 +49,16 @@ module.exports = function (app, options) {
       ctx.args.filter.include = include;
     }
 
+    if (RelUtils.isJSONAPIPagination(ctx.req.query)) {
+      ctx.res.set({'Content-Type': 'application/vnd.api+json'});
```

Comment:
Yup, agreed. This is being applied like a bandage all over the place. Probably worth:
- checking tests appropriately cover checking that the content type is set
- centralize setting application/vnd.api+json